Tests/Unit/Extend/EmailTest.php

```diff
@@ -66,4 +101,133 @@ public function testSendSupportEmailProvider()
             'complete' => [$aEmailDataComplete]
         ];
     }
+
+    public function testSendOrderEmailToUser()
+    {
+        $oOrderStub = $this->getMockBuilder(Order::class)
+            ->disableOriginalConstructor()
+            ->setMethods(['isCustomPaymentMethod', 'getOrderUser', 'getBasket', 'getPayment', '__get'])
+            ->getMock();
+
+        $oOrderStub->method('__get')
+            ->will(
+                $this->returnCallback(
+                    function ($sA) {
+                        return new Field($sA);
+                    })
+            );
+
+        $oOrderStub->method('isCustomPaymentMethod')
+            ->willReturn(true);
+
+        $oBasket = $this->getMockBuilder(Basket::class)
+            ->disableOriginalConstructor()
+            ->setMethods(['getBasketArticles'])
+            ->getMock();
+
+        $oOrderStub->method('getBasket')
+            ->willReturn($oBasket);
+
+        $oPaymentStub = $this->getMockBuilder(UserPayment::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $oPaymentStub->method('__get')
+            ->with('oxuserpayments__oxpaymentsid')
+            ->willReturn(new Field('oxempty'));
+
+        $oOrderStub->method('getPayment')
+            ->willReturn($oPaymentStub);
+
+        $oUserStub = $this->getMockBuilder(\OxidEsales\Eshop\Application\Model\User::class)
+            ->disableOriginalConstructor()
+            ->setMethods(['__get'])
+            ->getMock();
+
+        $oUserStub->method('__get')
+            ->will(
+                $this->returnCallback(
+                    function ($sA) {
+                        return new Field($sA);
+                    })
+            );
+
+        $oOrderStub->method('getOrderUser')
+            ->willReturn($oUserStub);
+
+        $oArticleStub = $this->getMockBuilder(Article::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $oBasket->method('getBasketArticles')
+            ->willReturn([$oArticleStub]);
+
+        $isSent = $this->_oEmail->sendOrderEmailToUser($oOrderStub, "my subject");
+
+        //this will end up in a flaky test depending on the email settings of the test server
```

Comment:
can u mock the email sending service?

Reply:
no, it is part of the OXID framework and they do not provide any functionality to set a mock mailer.